src/main/java/bookstore/model/Category.java

```diff
@@ -0,0 +1,36 @@
+package bookstore.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Table(name = "categories")
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = false")
+@Setter
+@Getter
+@RequiredArgsConstructor
```

Comment:
```suggestion
@NoArgsConstructor
```
RequiredArgsConstructor creates a constructor with final fields. In your case you should use NoArgsConstructor